cmake/modules/VersionInfo.cmake

```diff
@@ -1,5 +1,6 @@
 FIND_PACKAGE(Git)
 IF(GIT_FOUND AND NOT FORCE_VERSION)
+	SET(TRUST_VERSION TRUE) # Used for build type color-coding
```

Comment:
I believe this comment got buried in Discord, but we should be able to intelligently version these without new magic variables.

The command that we're issuing higher is:

```bash
git describe --tags --match v[0-9].[0-9].[0-9]*
```

So to see this in action, try this on a "tagged" release (on a separate branch please!)
```bash
git reset --hard v1.2.0 # reset to a tagged release
git describe --tags --match v[0-9].[0-9].[0-9]*
# outputs "v1.2.0"
```

Run the same command from the tip of a branch:
```bash
git pull # get the tip of the branch
git describe --tags --match v[0-9].[0-9].[0-9]*
# outputs "v1.2.0-495-gdb200fb90"
```
Now the next part of CMake is the important one... 

```cmake
STRING(REPLACE "-" ";" TAG_LIST "${GIT_TAG}")
```

What this is doing is it's placing a semicolon where a dash is.  This isn't obvious (and please feel free to add comments so it's more intuitive to others), but the `;` character is the special character in CMake to turn a string into a list/array.

So... now... it can be assumed that:

```cmake
IF(TAG_LIST_LENGTH EQUAL 1)
   #  we're probably a tagged release!
ENDIF()
```

The only downside to this parsing crap is that forks don't behave the same because their tags are missing (this is a pretty big problem and it's why we added the `FORCE_VERSION` flag).  For this, a bit more trickery may be needed, such as blindly trusting `FORCE_VERSION` as stable, or perhaps parsing `FORCE_VERSION` again.

Reply:
I've never used `cmake` before so I'm learning as I go here. I appreciate you pointing out a cleaner way to accomplish version auto-detection! I'll get to work on this.